tests/jballerina-unit-test/src/test/resources/test-src/statements/arrays/array_fill_runtime_test.bal

```diff
@@ -144,6 +144,56 @@ function tesTwoDimensionalArrayWithConstantSizeReferenceFill() {
     assertEqualPanic([1,2,3], b[1]);
 }
 
+function testAnonRecordTypeWithDefaultValueFill() {
+    record {| int i = 101; |}[3] testArray = [];
+    testArray[1] = {i: 202};
+    assertEqualPanic(3, testArray.length());
+    assertEqualPanic(101, testArray[0].i);
+    assertEqualPanic(202, testArray[1].i);
+    assertEqualPanic(101, testArray[2].i);
+
+    testAnonRecordArray(getAnonFromAnon, "Jane");
+    testAnonRecordArray(getAnonFromPerson, "default");
+    testAnonRecordArray(getPersonFromAnon, "John");
+    testAnonRecordArray(getPersonFromPerson, "default");
+}
+
+function testAnonRecordArray(function () returns record {|string name;|}[] func, string name) {
+    var resultArr = func();
+    resultArr[1] = {name: "Jack"};
+    assertEqualPanic(2, resultArr.length());
+    assertEqualPanic(name, resultArr[0].name);
+    assertEqualPanic("Jack", resultArr[1].name);    
+}
+
+type Person record {|
+    string name = "default";
+|};
+
+function getAnonFromAnon() returns record {| string name; |}[] {
+    record {|
+        string name = "Jane";
+    |}[] arr = [];
+    return arr;
+}
+
+function getAnonFromPerson() returns record {| string name; |}[] {
+    Person[] arr = [];
```

Comment:
Shall we add a test with Person & readonly array too

Reply:
As per spec, we don't have a filler value for intersection types. Therefore, `Person & readonly` will give an `IllegalListInsertion` error.
https://ballerina-platform.slack.com/archives/CAXS256RX/p1621243803134900

Reply:
The spec has been updated according to https://github.com/ballerina-platform/ballerina-spec/issues/850. Have we fixed this? Shall we create an issue if not?

Reply:
created https://github.com/ballerina-platform/ballerina-lang/issues/30977